packages/math/src/pool/concentrated/quotes.ts

```diff
@@ -0,0 +1,215 @@
+
+interface SwapState {
+  amountRemaining: Dec;
+  amountCalculated: Dec;
+  inittedTickIndex: number;
+  sqrtPrice: Dec;
+  liquidity: Dec;
+  feeGrowthGlobal: Dec;
+}
+
+function calcOutGivenIn({
+  tokenIn,
+  tokenDenom0,
+  poolLiquidity,
+  inittedTicksWithNetLiquidity,
+  curTick,
+  curSqrtPrice,
+  precisionFactorAtPriceOne,
+  swapFee,
+  priceLimit = new Dec(0),
+}: QuoteOutGivenInParams): Int {
+  const isZeroForOne = tokenIn.denom === tokenDenom0;
+  if (isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = minSpotPrice;
+  } else if (!isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = maxSpotPrice;
+  }
+
+  const sqrtPriceLimit = approxSqrt(priceLimit);
+  const swapStrategy = makeSwapStrategy(isZeroForOne, sqrtPriceLimit, swapFee);
+  const tokenInAmountSpecified = new Dec(tokenIn.amount);
+
+  const curTickWithNetLiq = inittedTicksWithNetLiquidity.find(({ tickIndex }) =>
+    tickIndex.equals(curTick)
+  );
+  if (!curTickWithNetLiq) {
+    throw new Error("curTickNet not found in inittedTicksWithNetLiquidity");
+  }
+  const curTickArrIndex =
+    inittedTicksWithNetLiquidity.indexOf(curTickWithNetLiq);
+  if (curTickArrIndex < 0) {
+    throw new Error(
+      "curTickWithNetLiq not found in inittedTicksWithNetLiquidity"
+    );
+  }
+
+  const swapState: SwapState = {
+    amountRemaining: tokenInAmountSpecified, // tokenIn
+    amountCalculated: new Dec(0), // tokenOut
+    inittedTickIndex: swapStrategy.initTickValue(curTickArrIndex),
+    sqrtPrice: curSqrtPrice,
+    liquidity: poolLiquidity,
+    feeGrowthGlobal: new Dec(0),
+  };
+
+  while (
+    swapState.amountRemaining.gt(smallestDec) &&
+    !swapState.sqrtPrice.equals(sqrtPriceLimit)
+  ) {
+    const nextTick: LiquidityDepth | undefined =
+      inittedTicksWithNetLiquidity?.[swapState.inittedTickIndex];
+    if (!nextTick) {
+      throw new TickOverflowError("Not enough ticks to calculate swap");
+    }
+
+    const nextTickSqrtPrice = tickToSqrtPrice(
+      nextTick.tickIndex,
+      precisionFactorAtPriceOne
+    );
+
+    const sqrtPriceTarget = swapStrategy.getSqrtTargetPrice(nextTickSqrtPrice);
+
+    const {
+      sqrtPriceNext,
+      amountInConsumed: amountOneIn,
+      amountOutComputed: amountZeroOut,
+      feeChargeTotal,
+    } = swapStrategy.computeSwapStepOutGivenIn(
+      swapState.sqrtPrice,
+      sqrtPriceTarget,
+      swapState.liquidity,
+      swapState.amountRemaining
+    );
+
+    swapState.sqrtPrice = sqrtPriceNext;
+    swapState.amountRemaining = swapState.amountRemaining.sub(
+      amountOneIn.add(feeChargeTotal)
+    );
+    swapState.amountCalculated = swapState.amountCalculated.add(amountZeroOut);
+
+    if (nextTickSqrtPrice.equals(sqrtPriceNext)) {
+      const liquidityNet = swapStrategy.setLiquidityDeltaSign(
+        new Dec(nextTick.netLiquidity.toString())
+      );
+
+      swapState.liquidity = addLiquidity(swapState.liquidity, liquidityNet);
+    }
+
+    swapState.inittedTickIndex++;
```

Comment:
This should be inside and at the end of this branch:
```
    if (nextTickSqrtPrice.equals(sqrtPriceNext)) {
      const liquidityNet = swapStrategy.setLiquidityDeltaSign(
        new Dec(nextTick.netLiquidity.toString())
      );

      swapState.liquidity = addLiquidity(swapState.liquidity, liquidityNet);
    }
```

The reason is that when `!nextTickSqrtPrice.equals(sqrtPriceNext)`, we are not reaching the next initialized tick yet so its value should stay the same

Reply:
I might have misinterpreted it as being outside. On another look now, the location looks good to me. Apologies for any confusion